File: Persistence/Source/Repositories/RejectedEntity.vb
Imports Wells.Model

Public Class RejectedEntity
    Public Enum RejectedReasons
        None
        DuplicatedId
        DuplicatedName
        WellNameEmpty
        WellNotFound
        FLNADepthGreaterThanWaterDepth
        DuplicatedDate
        Unknown
    End Enum

    ReadOnly Property Entity As IBusinessObject
    ReadOnly Property OriginalRow As Integer
    ReadOnly Property Reason As RejectedReasons

    Sub New(entity As IBusinessObject, originalRow As Integer, reason As RejectedReasons)
        Me.Entity = entity
        Me.OriginalRow = originalRow
        Me.Reason = reason
    End Sub

End Class

File: Persistence/CorePersistence/src/RejectedEntity.cs
using System.ComponentModel;
using Wells.StandardModel.Models;

namespace Wells.CorePersistence
{
    public class RejectedEntity
    {
        public IBusinessObject Entity { get; }
        public int OriginalRow { get; }
        public RejectedReasons Reason { get; }

        public RejectedEntity(IBusinessObject entity, int originalRow, RejectedReasons reason)
        {
            Entity = entity;
            OriginalRow = originalRow;
            Reason = reason;
        }
    }

    public enum RejectedReasons
    {
        [Description("Ninguna")] None,
        [Description("Id duplicado")] DuplicatedId,
        [Description("Nombre duplicado")] DuplicatedName,
        [Description("Pozo sin nombre")] WellNameEmpty,
        [Description("Pozo no encontrado")] WellNotFound,
        [Description("Profundidad de FLNA mayor al del agua")] FLNADepthGreaterThanWaterDepth,
        [Description("Fecha duplicada")] DuplicatedDate,
        [Description("Desconocido")] Unknown
    }
}
